Core/Src/main.c

```diff
@@ -175,385 +182,360 @@ void sendData(){
-
-  /*Configure GPIO pins : PC2 PC3 */
-  GPIO_InitStruct.Pin = GPIO_PIN_2|GPIO_PIN_3;
-  GPIO_InitStruct.Mode = GPIO_MODE_IT_RISING;
-  GPIO_InitStruct.Pull = GPIO_NOPULL;
-  HAL_GPIO_Init(GPIOC, &GPIO_InitStruct);
-
-  /*Configure GPIO pins : PB14 PB15 */
-  GPIO_InitStruct.Pin = GPIO_PIN_14|GPIO_PIN_15;
-  GPIO_InitStruct.Mode = GPIO_MODE_IT_RISING;
-  GPIO_InitStruct.Pull = GPIO_NOPULL;
-  HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);
-
-  /* EXTI interrupt init*/
-  HAL_NVIC_SetPriority(EXTI2_TSC_IRQn, 0, 0);
-  HAL_NVIC_EnableIRQ(EXTI2_TSC_IRQn);
-
-  HAL_NVIC_SetPriority(EXTI3_IRQn, 0, 0);
-  HAL_NVIC_EnableIRQ(EXTI3_IRQn);
-
-  HAL_NVIC_SetPriority(EXTI15_10_IRQn, 0, 0);
-  HAL_NVIC_EnableIRQ(EXTI15_10_IRQn);
-
-/* USER CODE BEGIN MX_GPIO_Init_2 */
-/* USER CODE END MX_GPIO_Init_2 */
+	GPIO_InitTypeDef GPIO_InitStruct = {0};
+	/* USER CODE BEGIN MX_GPIO_Init_1 */
+	/* USER CODE END MX_GPIO_Init_1 */
+
+	/* GPIO Ports Clock Enable */
+	__HAL_RCC_GPIOC_CLK_ENABLE();
+	__HAL_RCC_GPIOA_CLK_ENABLE();
+	__HAL_RCC_GPIOB_CLK_ENABLE();
+
+	/*Configure GPIO pins : PC2 PC3 */
+	GPIO_InitStruct.Pin = GPIO_PIN_2|GPIO_PIN_3;
+	GPIO_InitStruct.Mode = GPIO_MODE_IT_FALLING;
+	GPIO_InitStruct.Pull = GPIO_PULLUP;
+	HAL_GPIO_Init(GPIOC, &GPIO_InitStruct);
+
+	/*Configure GPIO pins : PB14 PB15 */
+	GPIO_InitStruct.Pin = GPIO_PIN_14|GPIO_PIN_15;
+	GPIO_InitStruct.Mode = GPIO_MODE_IT_FALLING;
+	GPIO_InitStruct.Pull = GPIO_PULLUP;
+	HAL_GPIO_Init(GPIOB, &GPIO_InitStruct);
+
+	/* EXTI interrupt init*/
+	HAL_NVIC_SetPriority(EXTI2_TSC_IRQn, 0, 0);
+	HAL_NVIC_EnableIRQ(EXTI2_TSC_IRQn);
+
+	HAL_NVIC_SetPriority(EXTI3_IRQn, 0, 0);
+	HAL_NVIC_EnableIRQ(EXTI3_IRQn);
+
+	HAL_NVIC_SetPriority(EXTI15_10_IRQn, 0, 0);
+	HAL_NVIC_EnableIRQ(EXTI15_10_IRQn);
+
+	/* USER CODE BEGIN MX_GPIO_Init_2 */
+	/* USER CODE END MX_GPIO_Init_2 */
 }
 
 /* USER CODE BEGIN 4 */
-void HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin)
-{
-	  UNUSED(GPIO_Pin);
-
-  if (GPIO_Pin == GPIO_PIN_2){
-	  state = HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_2);
-	  button = BUTTON_1;	//red
-  }
-  if (GPIO_Pin == GPIO_PIN_3){
-	  state = HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_3);
-	  button = BUTTON_2;	//green
-  }
-  if (GPIO_Pin == GPIO_PIN_14){
-	  state = HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_14);
-	  button = BUTTON_3;	//yellow
-  }
-  if (GPIO_Pin == GPIO_PIN_15){
-	  state = HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_15);
-	  button = BUTTON_4;	//blue
-  }
-  HAL_TIM_Base_Start_IT(&htim16);
-
-}
-
 void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef *htim){
 	 UNUSED(htim);
 
 	if(htim == &htim16){
-		if(HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_2) == state && button == BUTTON_1){
-			bitmask_set(0);
-		}
-		if(HAL_GPIO_ReadPin(GPIOC, GPIO_PIN_3) == state && button == BUTTON_2){
-			bitmask_set(1);
-		}
-		if(HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_14) == state && button == BUTTON_3){
-			bitmask_set(2);
-		}
-		if(HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_15) == state && button == BUTTON_4){
-			bitmask_set(3);
-		}
+		timer_flag = true;
```

Comment:
How come this code change is related to watchdog functionality?

Reply:
I deleted the use of the original timer for button debouncing so I reused that timer16 for the watchdog delay. Specifically, I use that flag to send the watchdog refresh after the timer is called back (ie 5 second delay into the program)